pyinter/interval.py

```diff
@@ -230,9 +230,33 @@ def union(self, other):
         else:
             return IntervalSet((self, other))
 
+    __or__ = __add__ = union
+
+    def __sub__(self, other):
+        """Returns a new Interval or an :class:`~pyinter.IntervalSet` representing the subtraction of this
+        :class:`~pyinter.Interval` with the other :class:`~pyinter.Interval`.
+
+        The result will contain everything that is contained by the left interval but not contained
+        by the second interval.
+
+        If the `other` interval is enclosed in this one then this will return a
+        :class:`~pyinter.IntervalSet`, otherwise this returns a :class:`~pyinter.Interval`.
+        """
+        if self in other:
+            return open(0, 0)
```

Comment:
There are test cases in #13. I didn't include them in this commit because they assert that the result is empty.